helper-scripts/backup_and_restore.sh

```diff
@@ -1,331 +1,670 @@
-        echo -e "\e[31mThe Architecture of the backed up mailcow OS is different then your restoring mailcow OS..."
-        sleep 2
-        echo -e "Skipping rspamd due to compatibility issues!\e[0m"
-      else
-        docker stop $(docker ps -qf name=rspamd-mailcow)
-        docker run -i --name mailcow-backup --rm \
-          -v ${RESTORE_LOCATION}:/backup:z \
-          -v $(docker volume ls -qf name=^${CMPS_PRJ}_rspamd-vol-1$):/rspamd:z \
-          ${DEBIAN_DOCKER_IMAGE} /bin/tar --use-compress-program="pigz -d -p ${THREADS}" -Pxvf /backup/backup_rspamd.tar.gz
-        docker start $(docker ps -aqf name=rspamd-mailcow)
-      fi
-      ;;
-    postfix)
-      docker stop $(docker ps -qf name=postfix-mailcow)
-      docker run -i --name mailcow-backup --rm \
-        -v ${RESTORE_LOCATION}:/backup:z \
-        -v $(docker volume ls -qf name=^${CMPS_PRJ}_postfix-vol-1$):/postfix:z \
-        ${DEBIAN_DOCKER_IMAGE} /bin/tar --use-compress-program="pigz -d -p ${THREADS}" -Pxvf /backup/backup_postfix.tar.gz
-      docker start $(docker ps -aqf name=postfix-mailcow)
-      ;;
-    mysql|mariadb)
-      SQLIMAGE=$(grep -iEo '(mysql|mariadb)\:.+' ${COMPOSE_FILE})
-      if [[ -z "${SQLIMAGE}" ]]; then
-        echo "Could not determine SQL image version, skipping restore..."
-        shift
-        continue
-      elif [ ! -f "${RESTORE_LOCATION}/mailcow.conf" ]; then
-        echo "Could not find the corresponding mailcow.conf in ${RESTORE_LOCATION}, skipping restore."
-        echo "If you lost that file, copy the last working mailcow.conf file to ${RESTORE_LOCATION} and restart the restore process."
-        shift
-        continue
-      else
-        read -p "mailcow will be stopped and the currently active mailcow.conf will be modified to use the DB parameters found in ${RESTORE_LOCATION}/mailcow.conf - do you want to proceed? [Y|n] " MYSQL_STOP_MAILCOW
-        if [[ ${MYSQL_STOP_MAILCOW,,} =~ ^(no|n|N)$ ]]; then
-          echo "OK, skipped."
+  for component in "${RESTORE_COMPONENTS[@]}"; do
+    echo
+    echo -e "\n${GREEN_COLOR}Restoring ${component}...${RESET}"
+    sleep 1
+    case ${component} in
+      vmail)
+        restore_docker_component "${RESTORE_LOCATION}" "${CMPS_PRJ}" "dovecot-mailcow" "vmail"
+
+        echo
+        echo "In most cases it is not required to run a full resync, you can run the command printed below at any time after testing wether the restore process broke a mailbox:"
+        echo
+        echo "docker exec $(docker ps -qf name=dovecot-mailcow) doveadm force-resync -A '*'"
+        echo
+
+        # Skip prompt if --yes flag is present
+        if [[ "${MAILCOW_YES_TO_ALL}" == "1" ]]; then
+          echo -e "${GREEN_COLOR}Forcing a resync now due to --yes flag.${RESET}"
+          docker exec $(docker ps -qf name=dovecot-mailcow) doveadm force-resync -A '*'
```

Comment:
I tested the restoration on my machine with the --yes flag. However, it seems that after the mails are restored doveadm has problems to resync them if the command to resync them is directly fired afterwards. If a user enters this manually (y to restore) it won't throw an error, i think dovecot is searching for the userdb file to early:

```
Forcing a resync now due to --yes flag.
Error: auth-master: userdb list: connect(/run/dovecot/auth-userdb) failed: Connection refused
Panic: file auth-master.c: line 436 (auth_master_unset_io): assertion failed: (conn->to == NULL)
```

Reply:
Then, should `sleep 1` ? or there's any other idea like making sure a file is exists and then resync ?

Reply:
Sleep 1 will mostlikely do it i guess, lets try that.

Reply:
Looks like this didn't fix it, make it 2seconds, now fixed.

Reply:
I tried several times, 2seconds solved it.